### ISSUE TITLE
[BUG] Stuttery auto-scroll to the bottom of the page when outputting code

### ISSUE THREAD

**Issue by conradg** [aws-samples/bedrock-claude-chat#447]
## Describe the bug

When writing long multi-line code blocks, the interface stutters a lot. 

**Filling out this field will help us investigate the issue efficiently. Providing detailed information allows us to set the appropriate priority. We appreciate your cooperation.**  

## To Reproduce

**Filling out this field will help us investigate the issue efficiently. Providing detailed information allows us to set the appropriate priority. We appreciate your cooperation.**  

1. Start a new chat
2. Ask a question that invokes a code response.

## screen recording

https://github.com/user-attachments/assets/438a05ca-8c48-4a72-8ef2-c5205533b667

## Additional context

It *seems* like this bug occurs more often on my 2nd monitor than on my my Macbook screen..

**Comment by k70suK3-k06a7ash1** [comment]
@conradg 
Is the version of the application you are checking the latest?

**Comment by k70suK3-k06a7ash1** [comment]
<img width="938" alt="スクリーンショット 2024-07-16 23 52 01" src="https://github.com/user-attachments/assets/1147688b-213a-4d04-ac13-6c74413971bc">

**Comment by conradg** [comment]
Hmm, no this is from a forked repo forked at this commit about 2 months ago.  https://github.com/aws-samples/bedrock-claude-chat/pull/303 

I can see a lot has happened since then, so I will try and upgrade to the latest changes

**Comment by k70suK3-k06a7ash1** [comment]
The following code is the relevant section
The useLayoutEffect is used so that the chat no longer bounces while it is being generated.

However, they are no longer followed during chat generation.

If you need this, modify it to scroll with delayed evaluation.

frontend/src/pages/ChatPage.tsx
```
  useLayoutEffect(() => {
    if (messages.length > 0) {
      scrollToBottom();
    } else {
      scrollToTop();
    }
  }, [messages, scrollToBottom, scrollToTop]);
```

**Comment by k70suK3-k06a7ash1** [comment]
@conradg 
https://github.com/aws-samples/bedrock-claude-chat/pull/450
Improved.
would you check this PR ?

**Comment by statefb** [comment]
Close issue because fixed by #450
